glib/src/param_spec.rs

```diff
@@ -384,6 +384,98 @@ macro_rules! define_param_spec_numeric {
     };
 }
 
+macro_rules! define_builder {
+    (@constructors $rust_type:ident, $builder_type:ident $(($($req_ident:ident: $req_ty:ty,)*))?) => {
+        impl<'a> $builder_type<'a> {
+            pub fn new(name: &'a str, $($($req_ident: $req_ty)*)?) -> Self {
```

Comment:
Maybe I missed that in one of the comments, but what's the plan for the macro here? It would use the `Default` impl of the builder and then panic at runtime if required parameters are not provided, or it would have some special syntax for required parameters (say `#[prop(@name = "bla", @other_required_parameter = blabla, optional_parameter = 123]`). The problem with the latter seems that it's not entirely obvious for the user which ones are required and which not, and the compiler errors will probably also not be the most useful.

Reply:
My current idea is
```
#[prop(get, set, builder(required_params).optional(opt_value))]
name: RefCell<String>
```
which should map to
```
ParamSpecString::builder(name, ...required_params)
  .optional(opt_value)
  .flags(READWRITE) // because the user specified get, set
```


To create a ParamSpecInt with a maximum of 5:
```
#[prop(get, set, builder().maximum(5))]
times: RefCell<i32>
```

Reply:
It can't panic, because builders have now safe constructors. I like my proposed syntax because it makes it clear you are calling the builder, therefore the user can just lookup the builder docs.

Your proposed solution
`#[prop(@name = "bla", @other_required_parameter = blabla, optional_parameter = 123]`
is not possible, because `optional_parameter = 123` may be a ParamSpec parameter specified by an external crate, conflicting with other attributes from the `Props` macro. Eg: an external crate may create a `ParamSpec` with the `get` parameter. 

Also, specifying the required parameters without order would make the builder implementation more difficult (We would need a ParamSpec*Args struct for each ParamSpec).

Reply:
> Eg: an external crate may create a ParamSpec with the get parameter.

I don't think there are that many external ParamSpec types, it would be a good idea to look at what are the available ones and figure out if the current use cases would fit properly there.

Reply:
Makes sense to me but is a bit strange syntax maybe :)